include/gz/common/DataFrame.hh

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+#ifndef GZ_COMMON_DATAFRAME_HH_
+#define GZ_COMMON_DATAFRAME_HH_
+
+#include <array>
+#include <numeric>
+#include <sstream>
+#include <string>
+#include <stdexcept>
+#include <unordered_map>
+#include <vector>
+
+#include <gz/common/CSVStreams.hh>
+#include <gz/common/Io.hh>
+
+#include <gz/math/TimeVaryingVolumetricGrid.hh>
```

Comment:
currently there are gz-common components that depend on gz-math, but the gz-common core library does not. We should consider moving the code in this PR to its own `csv` component or adding a math dependency to gz-common core

Reply:
this could also be added as a component of `gz-utils`

Reply:
Done in https://github.com/gazebosim/gz-common/pull/402/commits/84753af49393db913378e7aca02bf3f0d2778f61.